search_service/api/user.py

```diff
@@ -0,0 +1,66 @@
+from http import HTTPStatus
+from typing import Iterable, Any
+
+from flask_restful import Resource, fields, marshal_with, reqparse
+
+from search_service.proxy import get_proxy_client
+
+
+user_fields = {
+    "name": fields.String,
+    "first_name": fields.String,
+    "last_name": fields.String,
+    "team_name": fields.String,
+    "email": fields.String,
+    "manager_email": fields.String,
+    "github_username": fields.String,
+    "is_active": fields.Boolean,
+    "employee_type": fields.String,
+}
+
+search_user_results = {
+    "total_results": fields.Integer,
+    "results": fields.Nested(user_fields, default=[])
+}
+
+
+class SearchUserAPI(Resource):
+    """
+    Search Table API
+    """
+    USER_INDEX = 'user_search_index'
+
+    def __init__(self) -> None:
+        self.proxy = get_proxy_client()
+
+        self.parser = reqparse.RequestParser(bundle_errors=True)
+
+        self.parser.add_argument('query_term', required=True, type=str)
+        self.parser.add_argument('page_index', required=False, default=0, type=int)
+        self.parser.add_argument('index', required=False, default=SearchUserAPI.USER_INDEX, type=str)
+
+        super(SearchUserAPI, self).__init__()
+
+    @marshal_with(search_user_results)
+    def get(self) -> Iterable[Any]:
+        """
+        Fetch search results based on query_term.
+        :return: list of search results. List can be empty if query
+        doesn't match any result
+        """
+        args = self.parser.parse_args(strict=True)
+
+        try:
+
+            results = self.proxy.fetch_user_search_results(
+                query_term=args['query_term'],
+                page_index=args['page_index'],
+                index=args.get(args['index'])
+            )
+
+            return results, HTTPStatus.OK
+
+        except RuntimeError:
```

Comment:
Why it is RuntimeError?

Reply:
It means the error/exception is thrown underneath which we want to return 500 back to FE.